backend/python/app/rest/entity_routes.py

```diff
@@ -44,11 +60,16 @@ def get_entities():
 def get_entity(id):
     try:
         result = entity_service.get_entity(id)
+        content_type = request.mimetype
     except Exception as e:
         error_message = getattr(e, "message", None)
         return jsonify({"error": (error_message if error_message else str(e))}), 500
 
     # HTTP status code 200 means OK
+    if content_type == "text/csv":
+        result = entity_service.get_entities()
+        return jsonify(generate_csv_from_list(result)), 200
+
```

Comment:
Sorry for the misunderstanding but we don't need CSV functionality for single entities, just the bulk GET request above is enough :)